### ISSUE TITLE
[JENKINS-53129] Parallel sequential stages in Blue Ocean showing wrong status

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/blueocean-plugin#4831]
<p>After updating the Blue ocean to the latest version 1.8.2 released 16 August, Blue Ocean is having issues displaying the status of parallel stages properly.</p>

<p>I have 2 stages in parallel which contains several sequential stage each. <br/>
 While they are <font color="#ff0000">both processing</font>, one of the parallel stages is showing as <b>green</b> (<font color="#00875a"><b>DONE</b></font>) and the other shown as Processing but not displaying the triggered builds.</p>

<p>You can find my pipeline code below.</p>

<pre>
 pipeline {
  agent any
  stages { stage(<span class="code-quote">'Pushing 2 Tests'</span>) {
    <span class="code-comment">//failFast <span class="code-keyword">true</span>
</span>    parallel {     
     stage(<span class="code-quote">'TE1'</span>) {
      stages {
       stage(<span class="code-quote">'PUSH TE1'</span>) {
steps {
 script {    
  <span class="code-comment">//Trigger a job
</span>  sleep time: 5, unit: <span class="code-quote">'MINUTES'</span>
  echo <span class="code-quote">"Test1"</span>
 }
}
       }
       stage(<span class="code-quote">'Archive Logs'</span>) {
steps {
 script {
  echo <span class="code-quote">"Archive Test1 Logs"</span>       
 }
}
       }
       stage(<span class="code-quote">'Insert Metrics'</span>) {
steps {
 script {
  echo <span class="code-quote">"Insert Test1 Metrics in DB"</span>
 }
}
       }
       stage(<span class="code-quote">'Compare Metrics'</span>) {
steps {
 script {
  echo <span class="code-quote">"Compare Metrics Test1 vs REF"</span>
 }
}
       }
      }
     }     
     stage(<span class="code-quote">'TE2'</span>) {
      stages {
       stage(<span class="code-quote">'PUSH TE2'</span>) {
steps {
 script {
  <span class="code-comment">//Trigger a job
</span>  sleep time: 5, unit: <span class="code-quote">'MINUTES'</span>
  echo <span class="code-quote">"Test2"</span>
 }
}
       }
       stage(<span class="code-quote">'Archive Logs'</span>) {
steps {
 script {
  echo <span class="code-quote">"Archive Test2 Logs"</span>
 }
}
       }
       stage(<span class="code-quote">'Insert Metrics'</span>) {
steps {
 script {
  echo <span class="code-quote">"Insert Test2 Metrics in DB"</span>
 }
}
       }
       stage(<span class="code-quote">'Compare Metrics'</span>) {
steps {
 script {
  echo <span class="code-quote">"Compare Metrics Test2 vs REF"</span>
 }
}
       }
      }
     }    
    }
   }  
}
}
</pre>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=ekassis">ekassis</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-53129" target="_blank">Parallel sequential stages in Blue Ocean showing wrong status</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/olamy.jpg" title="olamy's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=olamy">olamy</a>
<li><b>status</b>: Closed
<li><b>priority</b>: Critical
<li><b>component(s)</b>: blueocean-plugin
<li><b>label(s)</b>: blueocean
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2018-09-19T06:57:34+00:00
<li><b>votes</b>: 1
<li><b>watchers</b>: 6
<li><b>imported</b>: 2025-12-01
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>After updating the Blue ocean to the latest version 1.8.2 released 16 August, Blue Ocean is having issues displaying the status of parallel stages properly.</p>

<p>I have 2 stages in parallel which contains several sequential stage each. 
 While they are <font color="#ff0000">both processing</font>, one of the parallel stages is showing as <b>green</b> (<font color="#00875a"><b>DONE</b></font>) and the other shown as Processing but not displaying the triggered builds.</p>

<p>You can find my pipeline code below.</p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java"> pipeline {
  agent any
  stages { stage(<span class="code-quote">'Pushing 2 Tests'</span>) {
    <span class="code-comment">//failFast <span class="code-keyword">true</span>
</span>    parallel {     
     stage(<span class="code-quote">'TE1'</span>) {
      stages {
       stage(<span class="code-quote">'PUSH TE1'</span>) {
        steps {
         script {    
          <span class="code-comment">//Trigger a job
</span>          sleep time: 5, unit: <span class="code-quote">'MINUTES'</span>        
          echo <span class="code-quote">"Test1"</span>        
         }
        }
       }
       stage(<span class="code-quote">'Archive Logs'</span>) {
        steps {
         script {
          echo <span class="code-quote">"Archive Test1 Logs"</span>       
         }
        }
       }
       stage(<span class="code-quote">'Insert Metrics'</span>) {
        steps {
         script {
          echo <span class="code-quote">"Insert Test1 Metrics in DB"</span>
         }
        }
       }
       stage(<span class="code-quote">'Compare Metrics'</span>) {
        steps {
         script {
          echo <span class="code-quote">"Compare Metrics Test1 vs REF"</span>
         }
        }
       }
      }
     }     
     stage(<span class="code-quote">'TE2'</span>) {
      stages {
       stage(<span class="code-quote">'PUSH TE2'</span>) {
        steps {
         script {
          <span class="code-comment">//Trigger a job
</span>          sleep time: 5, unit: <span class="code-quote">'MINUTES'</span>
          echo <span class="code-quote">"Test2"</span>
         }
        }
       }
       stage(<span class="code-quote">'Archive Logs'</span>) {
        steps {
         script {
          echo <span class="code-quote">"Archive Test2 Logs"</span>
         }
        }
       }
       stage(<span class="code-quote">'Insert Metrics'</span>) {
        steps {
         script {
          echo <span class="code-quote">"Insert Test2 Metrics in DB"</span>
         }
        }
       }
       stage(<span class="code-quote">'Compare Metrics'</span>) {
        steps {
         script {
          echo <span class="code-quote">"Compare Metrics Test2 vs REF"</span>
         }
        }
       }
      }
     }    
    }
   }  
}
}
</pre>
</div></div></pre>
</details>
</details>
<details><summary><i>environment</i></summary>

```
Jenkins version:  2.121.3<br/>
Blue Ocean version: 1.8.2<br/>
OS: RHEL7.2<br/>
Browser: Google Chrome
```
</details>
<details><summary><i>8 attachments</i></summary>

- [image-2018-08-20-12-43-44-625.png](https://issues.jenkins.io/secure/attachment/43793/image-2018-08-20-12-43-44-625.png)
  > ![image-2018-08-20-12-43-44-625.png](https://issues.jenkins.io/secure/attachment/43793/image-2018-08-20-12-43-44-625.png)
- [image-2018-08-20-19-51-36-113.png](https://issues.jenkins.io/secure/attachment/43796/image-2018-08-20-19-51-36-113.png)
  > ![image-2018-08-20-19-51-36-113.png](https://issues.jenkins.io/secure/attachment/43796/image-2018-08-20-19-51-36-113.png)
- [image-2018-08-20-19-53-40-887.png](https://issues.jenkins.io/secure/attachment/43797/image-2018-08-20-19-53-40-887.png)
  > ![image-2018-08-20-19-53-40-887.png](https://issues.jenkins.io/secure/attachment/43797/image-2018-08-20-19-53-40-887.png)
- [image-2018-08-22-16-11-36-340.png](https://issues.jenkins.io/secure/attachment/43824/image-2018-08-22-16-11-36-340.png)
  > ![image-2018-08-22-16-11-36-340.png](https://issues.jenkins.io/secure/attachment/43824/image-2018-08-22-16-11-36-340.png)
- [image-2018-08-23-01-15-45-479.png](https://issues.jenkins.io/secure/attachment/43833/image-2018-08-23-01-15-45-479.png)
  > ![image-2018-08-23-01-15-45-479.png](https://issues.jenkins.io/secure/attachment/43833/image-2018-08-23-01-15-45-479.png)
- [image-2018-09-22-13-17-53-432.png](https://issues.jenkins.io/secure/attachment/44566/image-2018-09-22-13-17-53-432.png)
  > ![image-2018-09-22-13-17-53-432.png](https://issues.jenkins.io/secure/attachment/44566/image-2018-09-22-13-17-53-432.png)
- [image-2018-09-23-19-13-12-757.png](https://issues.jenkins.io/secure/attachment/44568/image-2018-09-23-19-13-12-757.png)
  > ![image-2018-09-23-19-13-12-757.png](https://issues.jenkins.io/secure/attachment/44568/image-2018-09-23-19-13-12-757.png)
- [image-2018-09-23-20-42-44-112.png](https://issues.jenkins.io/secure/attachment/44569/image-2018-09-23-20-42-44-112.png)
  > ![image-2018-09-23-20-42-44-112.png](https://issues.jenkins.io/secure/attachment/44569/image-2018-09-23-20-42-44-112.png)
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/vivek.png" title="vivek's avatar" /> vivek:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-53129?focusedId=346629&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-346629">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>&gt; the other shown as Processing but not displaying the triggered builds.</p>

<p>You mean when you click them the steps are not shown?</p>

<p> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=nicu" class="user-hover" rel="nicu">nicu</a>  PTAL</p>
</pre>
</details>

</ul>
</details>

<p>> the other shown as Processing but not displaying the triggered builds.</p>

<p>You mean when you click them the steps are not shown?</p>

<p> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=nicu" class="user-hover" rel="nicu">nicu</a>  PTAL</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>ekassis:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-53129?focusedId=346630&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-346630">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=vivek" class="user-hover" rel="vivek">vivek</a> Yes indeed, when you click on the stage shown as <font color="#0747a6"><b>Processing</b></font><font color="#172b4d">, the steps are not shown. </font>
As for the other <ins>processing</ins> stage shown <font color="#de350b"><b>wrongly</b> </font>as <font color="#00875a"><b>Passed</b></font>, its steps are shown and can be selected. </p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=vivek" class="user-hover" rel="vivek">vivek</a> Yes indeed, when you click on the stage shown as <font color="#0747a6"><b>Processing</b></font><font color="#172b4d">, the steps are not shown. </font><br/>
As for the other <ins>processing</ins> stage shown <font color="#de350b"><b>wrongly</b> </font>as <font color="#00875a"><b>Passed</b></font>, its steps are shown and can be selected. </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/vivek.png" title="vivek's avatar" /> vivek:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-53129?focusedId=346632&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-346632">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Sometimes when stage is in progress, steps might not appear instantly because it might not yet be running so api serving them for the selected stage still be returning no steps. </p>

<p>Regarding wrong status, <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=olamy" class="user-hover" rel="olamy">olamy</a> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=nicu" class="user-hover" rel="nicu">nicu</a> it might be api not returning correct status? I think there is existing issue with status of parallel stages, there might be ticket some where or may be its new one. </p>
</pre>
</details>

</ul>
</details>

<p>Sometimes when stage is in progress, steps might not appear instantly because it might not yet be running so api serving them for the selected stage still be returning no steps. </p>

<p>Regarding wrong status, <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=olamy" class="user-hover" rel="olamy">olamy</a> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=nicu" class="user-hover" rel="nicu">nicu</a> it might be api not returning correct status? I think there is existing issue with status of parallel stages, there might be ticket some where or may be its new one. </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>ekassis:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-53129?focusedId=346637&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-346637">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Sorry <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=vivek" class="user-hover" rel="vivek">vivek</a>, but this is not the case on my side my 2 stages are processing exactly the same jobs that takes 2 hours each, we are not talking about instantly status appearance. 
 As shown in below screenshots, the processing stage that is showing wrong Passed status (PUSH TE2), is showing the triggered builds properly. As for the stage showing processing status (PUSH TE1), the Triggered Builds section does not exist.</p>

<p><span class="image-wrap" style=""><img src="https://issues.jenkins.io/secure/attachment/43796/43796_image-2018-08-20-19-51-36-113.png" height="269" width="410" style="border: 0px solid black" /></span><span class="image-wrap" style=""><img src="https://issues.jenkins.io/secure/attachment/43797/43797_image-2018-08-20-19-53-40-887.png" height="247" width="501" style="border: 0px solid black" /></span></p>
</pre>
</details>

</ul>
</details>

<p>Sorry <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=vivek" class="user-hover" rel="vivek">vivek</a>, but this is not the case on my side my 2 stages are processing exactly the same jobs that takes 2 hours each, we are not talking about instantly status appearance. <br/>
 As shown in below screenshots, the processing stage that is showing wrong Passed status (PUSH TE2), is showing the triggered builds properly. As for the stage showing processing status (PUSH TE1), the Triggered Builds section does not exist.</p>

<p><span class="image-wrap" style=""><img src="https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-last/refs/heads/main/attachments/43796/image-2018-08-20-19-51-36-113.png" height="269" width="410" style="border: 0px solid black" /></span><span class="image-wrap" style=""><img src="https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/43797/image-2018-08-20-19-53-40-887.png" height="247" width="501" style="border: 0px solid black" /></span></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>ekassis:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-53129?focusedId=346638&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-346638">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=vivek" class="user-hover" rel="vivek">vivek</a> As seen in my previous comment, when clicking on the upper stage it is giving me the details of the Parent stage (<b><font color="#de350b"><em>Pushing 2 Tests / TE1</em></font></b>) instead of the child (<b><em>PUSH TE1</em></b>) and when selecting the bottom stage it is giving me the details of the right stage (<font color="#00875a"><b><em>PUSH TE2</em></b></font>) but with <font color="#de350b"><ins><b>wrong status</b></ins></font>.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=vivek" class="user-hover" rel="vivek">vivek</a> As seen in my previous comment, when clicking on the upper stage it is giving me the details of the Parent stage (<b><font color="#de350b"><em>Pushing 2 Tests / TE1</em></font></b>) instead of the child (<b><em>PUSH TE1</em></b>) and when selecting the bottom stage it is giving me the details of the right stage (<font color="#00875a"><b><em>PUSH TE2</em></b></font>) but with <font color="#de350b"><ins><b>wrong status</b></ins></font>.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>ekassis:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-53129?focusedId=346641&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-346641">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=vivek" class="user-hover" rel="vivek">vivek</a>, as a summary the expected behavior is to show processing stages as <font color="#0747a6"><b>Processing</b></font> and to be able to access the <b>Triggered Builds</b> upon clicking on the stage.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=vivek" class="user-hover" rel="vivek">vivek</a>, as a summary the expected behavior is to show processing stages as <font color="#0747a6"><b>Processing</b></font> and to be able to access the <b>Triggered Builds</b> upon clicking on the stage.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>ekassis:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-53129?focusedId=346875&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-346875">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=vivek" class="user-hover" rel="vivek">vivek</a>, in addition currently the Processing stages are not visible on Blue Ocean. They only appear once they are done.
<span class="image-wrap" style=""><img src="https://issues.jenkins.io/secure/attachment/43824/43824_image-2018-08-22-16-11-36-340.png" height="261" width="815" style="border: 0px solid black" /></span></p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=vivek" class="user-hover" rel="vivek">vivek</a>, in addition currently the Processing stages are not visible on Blue Ocean. They only appear once they are done.<br/>
<span class="image-wrap" style=""><img src="https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-last/refs/heads/main/attachments/43824/image-2018-08-22-16-11-36-340.png" height="261" width="815" style="border: 0px solid black" /></span></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>ekassis:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-53129?focusedId=346882&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-346882">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=vivek" class="user-hover" rel="vivek">vivek</a>, <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=nicu" class="user-hover" rel="nicu">nicu</a>, please let me know if you need more details.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=vivek" class="user-hover" rel="vivek">vivek</a>, <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=nicu" class="user-hover" rel="nicu">nicu</a>, please let me know if you need more details.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>ekassis:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-53129?focusedId=346941&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-346941">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=vivek" class="user-hover" rel="vivek">vivek</a>, another case where 2 parallel stages are processing but shown in Blue Ocean in Grey (Pending).</p>

<p><span class="image-wrap" style=""><img src="https://issues.jenkins.io/secure/attachment/43833/43833_image-2018-08-23-01-15-45-479.png" height="283" width="319" style="border: 0px solid black" /></span></p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=vivek" class="user-hover" rel="vivek">vivek</a>, another case where 2 parallel stages are processing but shown in Blue Ocean in Grey (Pending).</p>

<p><span class="image-wrap" style=""><img src="https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/43833/image-2018-08-23-01-15-45-479.png" height="283" width="319" style="border: 0px solid black" /></span></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>ekassis:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-53129?focusedId=347446&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-347446">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=nicu" class="user-hover" rel="nicu">nicu</a> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=vivek" class="user-hover" rel="vivek">vivek</a> is there any estimation when this issue will be fixed?</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=nicu" class="user-hover" rel="nicu">nicu</a> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=vivek" class="user-hover" rel="vivek">vivek</a> is there any estimation when this issue will be fixed?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/olamy.jpg" title="olamy's avatar" /> olamy:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-53129?focusedId=349288&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-349288">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>related to <a href="https://issues.jenkins.io/browse/JENKINS-53311" title="[Blue Ocean] Sequential stages appear to be completed but are still executing" class="issue-link" data-issue-key="JENKINS-53311"><del>JENKINS-53311</del></a> ?</p>
</pre>
</details>

</ul>
</details>

<p>related to <a href="https://issues.jenkins.io/browse/JENKINS-53311" title="[Blue Ocean] Sequential stages appear to be completed but are still executing" class="issue-link" data-issue-key="JENKINS-53311"><del>JENKINS-53311</del></a> ?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>ekassis:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-53129?focusedId=349305&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-349305">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=olamy" class="user-hover" rel="olamy">olamy</a> yes related to <a href="https://issues.jenkins.io/browse/JENKINS-53311" title="[Blue Ocean] Sequential stages appear to be completed but are still executing" class="issue-link" data-issue-key="JENKINS-53311"><del>JENKINS-53311</del></a> but with additional problem in my Jira related to "Triggered Builds" Field.</p>

<p>In addition my Jira issue was reported before <a href="https://issues.jenkins.io/browse/JENKINS-53311" title="[Blue Ocean] Sequential stages appear to be completed but are still executing" class="issue-link" data-issue-key="JENKINS-53311"><del>JENKINS-53311</del></a> and contains more examples </p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=olamy" class="user-hover" rel="olamy">olamy</a> yes related to <a href="https://issues.jenkins.io/browse/JENKINS-53311" title="[Blue Ocean] Sequential stages appear to be completed but are still executing" class="issue-link" data-issue-key="JENKINS-53311"><del>JENKINS-53311</del></a> but with additional problem in my Jira related to "Triggered Builds" Field.</p>

<p>In addition my Jira issue was reported before <a href="https://issues.jenkins.io/browse/JENKINS-53311" title="[Blue Ocean] Sequential stages appear to be completed but are still executing" class="issue-link" data-issue-key="JENKINS-53311"><del>JENKINS-53311</del></a> and contains more examples </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/olamy.jpg" title="olamy's avatar" /> olamy:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-53129?focusedId=349490&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-349490">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>should be fixed with <a href="https://issues.jenkins.io/browse/JENKINS-53311" title="[Blue Ocean] Sequential stages appear to be completed but are still executing" class="issue-link" data-issue-key="JENKINS-53311"><del>JENKINS-53311</del></a> </p>
</pre>
</details>

</ul>
</details>

<p>should be fixed with <a href="https://issues.jenkins.io/browse/JENKINS-53311" title="[Blue Ocean] Sequential stages appear to be completed but are still executing" class="issue-link" data-issue-key="JENKINS-53311"><del>JENKINS-53311</del></a> </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>ekassis:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-53129?focusedId=349491&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-349491">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Thank you <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=olamy" class="user-hover" rel="olamy">olamy</a>.</p>

<p>When it is expected to have the plugin patch ready in Jenkins ?</p>
</pre>
</details>

</ul>
</details>

<p>Thank you <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=olamy" class="user-hover" rel="olamy">olamy</a>.</p>

<p>When it is expected to have the plugin patch ready in Jenkins ?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/olamy.jpg" title="olamy's avatar" /> olamy:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-53129?focusedId=349494&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-349494">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=ekassis" class="user-hover" rel="ekassis">ekassis</a> Not sure. People are a bit busy this week with JW.</p>

<p> </p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=ekassis" class="user-hover" rel="ekassis">ekassis</a> Not sure. People are a bit busy this week with JW.</p>

<p> </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>ekassis:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-53129?focusedId=349749&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-349749">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=olamy" class="user-hover" rel="olamy">olamy</a> the status issue was resolved with your last fix (Blue Ocean 1.8.3)(Jenkins 2.138.1), but the <font color="#de350b"><b>Triggered Builds</b></font> Section is not appearing when having processing stage.
The Triggered Build section is only appearing when the stage is done.</p>

<p>In addition when having several parallel stages processing at the same time, only 1 of the stages is having the blue circle with the blue dot blinking inside.</p>

<p>Can you please check?</p>

<p><span class="image-wrap" style=""><img src="https://issues.jenkins.io/secure/attachment/44566/44566_image-2018-09-22-13-17-53-432.png" height="237" width="465" style="border: 0px solid black" /></span></p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=olamy" class="user-hover" rel="olamy">olamy</a> the status issue was resolved with your last fix (Blue Ocean 1.8.3)(Jenkins 2.138.1), but the <font color="#de350b"><b>Triggered Builds</b></font> Section is not appearing when having processing stage.<br/>
The Triggered Build section is only appearing when the stage is done.</p>

<p>In addition when having several parallel stages processing at the same time, only 1 of the stages is having the blue circle with the blue dot blinking inside.</p>

<p>Can you please check?</p>

<p><span class="image-wrap" style=""><img src="https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-last/refs/heads/main/attachments/44566/image-2018-09-22-13-17-53-432.png" height="237" width="465" style="border: 0px solid black" /></span></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/olamy.jpg" title="olamy's avatar" /> olamy:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-53129?focusedId=349751&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-349751">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><b>Triggered Builds but</b> how do you trigger the jobs?</p>
</pre>
</details>

</ul>
</details>

<p><b>Triggered Builds but</b> how do you trigger the jobs?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>ekassis:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-53129?focusedId=349752&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-349752">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I trigger jobs from inside the stage as shown below code sample.
 Noting that the <font color="#de350b"><b>Triggered Builds</b></font> section was working properly previously before you started allowing in Blue Ocean to display sequential stages inside a parallel one.</p>

<p><ins>It is very important and crucial to see the Triggered Builds within a stage while PROCESSING and not just when the stage is DONE.</ins></p>

<p> </p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">stage(<span class="code-quote">'Pushing 2 Tests and 2 Debug jobs'</span>) {
    <span class="code-comment">//failFast <span class="code-keyword">true</span>
</span>    when {
     expression {
      Push2TEs == <span class="code-quote">'<span class="code-keyword">true</span>'</span>
     }
    }
    parallel {     stage(<span class="code-quote">'TE1'</span>) {
      stages {
       stage(<span class="code-quote">'PUSH TE1'</span>) {
        options {
         retry(2)
         timeout(time: 3, unit: <span class="code-quote">'HOURS'</span>)
        }
        steps {
         script {
          timestamps {
           println(<span class="code-quote">"${STAGE_BEGIN} PUSH TE1"</span>)
           def TE1build = build job: <span class="code-quote">"TESTING/${TEST_LABEL1}/PAC_PTE/0010.TESTING-Headline"</span>,
            parameters: [
             [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'MX_VERSION'</span>, value: MX_VERSION],
             [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'MX_CHANGELIST'</span>, value: MX_CHANGELIST],
             [$class: <span class="code-quote">'BooleanParameterValue'</span>, name: <span class="code-quote">'Execute_benchmark'</span>, value: <span class="code-keyword">true</span>],
             [$class: <span class="code-quote">'BooleanParameterValue'</span>, name: <span class="code-quote">'Generate_traceability_reports'</span>, value: <span class="code-keyword">true</span>],
             [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'SETUPS'</span>, value: <span class="code-quote">'Custom'</span>],
             [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'Custom_Script_Before'</span>, value: <span class="code-quote">'/nfs_qa_pac/pac-pte/jenkins/getEnvProperties.sh'</span>],
             [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'Custom_Script_After'</span>, value: <span class="code-quote">'/nfs_qa_pac/pac-pte/jenkins/restoreEnv.sh'</span>]
            ],
            propagate: <span class="code-keyword">false</span>
           LogJobStatus(<span class="code-quote">"${TEST_LABEL1}"</span>, TE1build.result)
           currentBuild.result = TE1build.result
          }
         }
        }
       }
       stage(<span class="code-quote">'Archive Logs'</span>) {
        steps {
         script {
          timestamps {
           println(<span class="code-quote">"${STAGE_BEGIN} Archive ${TEST_LABEL1} Logs"</span>)
           <span class="code-comment">//sleep time: 1, unit: <span class="code-quote">'MINUTES'</span>
</span>           def archiveLogsTE1 = build job: <span class="code-quote">'Archiving'</span>,
            parameters: [
             [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'BENCH_MARK_NAME'</span>, value: TE1_BENCH_MARK_NAME],
             [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'VERSION'</span>, value: MX_VERSION],
             [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'CHANGELIST'</span>, value: MX_CHANGELIST],
             [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'APP_DIR'</span>, value: TE1_APP_DIR],
             [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'RESULTS_DIR'</span>, value: TE1_LOG_DIR],
             [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'KPI_DIR'</span>, value: TE1_WORKSPACE],
             [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'TPK'</span>, value: TPK],
             [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'JOB_ID'</span>, value: JOB_ID],
             [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'TEST_ID'</span>, value: TEST_LABEL1],
             [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'NODE'</span>, value: TE1_NODE]
            ],
            propagate: <span class="code-keyword">false</span>
           currentBuild.result = archiveLogsTE1.result
           LogJobStatus(<span class="code-quote">"Archive ${TEST_LABEL1} Logs"</span>, archiveLogsTE1.result)           println(<span class="code-quote">"${STAGE_END} Archive ${TEST_LABEL1} Logs"</span>)
          }
         }
        }
       }
      }
     }     stage(<span class="code-quote">'TE2'</span>) {
      stages {
       stage(<span class="code-quote">'PUSH TE2'</span>) {
        options {
         retry(2)
         timeout(time: 3, unit: <span class="code-quote">'HOURS'</span>)
        }
        steps {
         script {
          timestamps {
           println(<span class="code-quote">"${STAGE_BEGIN} PUSH TE2"</span>)
           def TE2build = build job: <span class="code-quote">"TESTING/${TEST_LABEL2}/PAC_PTE/0010.TESTING-Headline"</span>,
            parameters: [
             [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'MX_VERSION'</span>, value: MX_VERSION],
             [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'MX_CHANGELIST'</span>, value: MX_CHANGELIST],
             [$class: <span class="code-quote">'BooleanParameterValue'</span>, name: <span class="code-quote">'Execute_benchmark'</span>, value: <span class="code-keyword">true</span>],
             [$class: <span class="code-quote">'BooleanParameterValue'</span>, name: <span class="code-quote">'Generate_traceability_reports'</span>, value: <span class="code-keyword">true</span>],
             [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'SETUPS'</span>, value: <span class="code-quote">'Custom'</span>],
             [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'Custom_Script_Before'</span>, value: <span class="code-quote">'/nfs_qa_pac/pac-pte/jenkins/getEnvProperties.sh'</span>],
             [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'Custom_Script_After'</span>, value: <span class="code-quote">'/nfs_qa_pac/pac-pte/jenkins/restoreEnv.sh'</span>]
            ],
            propagate: <span class="code-keyword">false</span>
           LogJobStatus(<span class="code-quote">"PUSH TE2"</span>, TE2build.result)
           currentBuild.result = TE2build.result       
          }
         }
        }
       }
       stage(<span class="code-quote">'Archive Logs'</span>) {
        steps {
         script {
          timestamps {
           println(<span class="code-quote">"${STAGE_BEGIN} Archive ${TEST_LABEL2} Logs"</span>)
           <span class="code-comment">//sleep time: 1, unit: <span class="code-quote">'MINUTES'</span>
</span>           def archiveLogsTE2 = build job: <span class="code-quote">'Archiving'</span>,
            parameters: [
             [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'BENCH_MARK_NAME'</span>, value: TE2_BENCH_MARK_NAME],
             [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'VERSION'</span>, value: MX_VERSION],
             [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'CHANGELIST'</span>, value: MX_CHANGELIST],
             [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'APP_DIR'</span>, value: TE2_APP_DIR],
             [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'RESULTS_DIR'</span>, value: TE2_LOG_DIR],
             [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'TPK'</span>, value: TPK],
             [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'JOB_ID'</span>, value: JOB_ID],
             [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'TEST_ID'</span>, value: TEST_LABEL2],
             [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'NODE'</span>, value: TE2_NODE]
            ],
            propagate: <span class="code-keyword">false</span>
           currentBuild.result = archiveLogsTE2.result
           LogJobStatus(<span class="code-quote">"Archive ${TEST_LABEL2} Logs"</span>, archiveLogsTE2.result)           println(<span class="code-quote">"${STAGE_END} Archive ${TEST_LABEL2} Logs"</span>)
          }
         }
        }
       }
      }
     }     stage(<span class="code-quote">'HEAD DEBUG'</span>) {
      stages {
       stage(<span class="code-quote">'PUSH HEAD DEBUG'</span>) {
        options {
         retry(2)
         timeout(time: 2, unit: <span class="code-quote">'HOURS'</span>)
        }
        steps {
         script {
          timestamps {
           println(<span class="code-quote">"${STAGE_BEGIN} PUSH HEAD DEBUG"</span>)
           <span class="code-keyword">try</span> {
            <span class="code-comment">//STREAM_ID=<span class="code-quote">'DEBUG230820181643'</span>
</span>            def pushHeadDebugJob = build job: <span class="code-quote">"TESTING/${STREAM_ID}/PAC_PTE/0010.TESTING-Headline"</span>,
             parameters: [
              [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'MX_VERSION'</span>, value: MX_VERSION],
              [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'MX_CHANGELIST'</span>, value: MX_CHANGELIST],
              [$class: <span class="code-quote">'BooleanParameterValue'</span>, name: <span class="code-quote">'Execute_benchmark'</span>, value: <span class="code-keyword">false</span>],
              [$class: <span class="code-quote">'BooleanParameterValue'</span>, name: <span class="code-quote">'Generate_traceability_reports'</span>, value: <span class="code-keyword">false</span>],
              [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'SETUPS'</span>, value: <span class="code-quote">'Custom'</span>],
              [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'Custom_Script_Before'</span>, value: <span class="code-quote">'/nfs_qa_pac/pac-pte/jenkins/getEnvProperties.sh'</span>],
              [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'Custom_Script_After'</span>, value: <span class="code-quote">'/nfs_qa_pac/pac-pte/jenkins/restoreEnv.sh'</span>]
             ],
             propagate: <span class="code-keyword">false</span>
            currentBuild.result = pushHeadDebugJob.result           } <span class="code-keyword">catch</span> (all) {
            echo <span class="code-quote">'Pushing Head Debug job Failed'</span>
            <span class="code-keyword">return</span> 0
           }
           println(<span class="code-quote">"${STAGE_END} PUSH HEAD DEBUG"</span>)
          }
         }
        }
       }
       stage(<span class="code-quote">'DEPLOY CLIENT HEAD'</span>) {
        steps {
         script {
          timestamps {
           println(<span class="code-quote">"${STAGE_BEGIN} Deploying Client HEAD"</span>)
           <span class="code-keyword">try</span> {
            def DeployClient = build job: <span class="code-quote">'DeployClient'</span>,
             parameters: [
              [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'NODE'</span>, value: <span class="code-quote">'pacpte_pte02nt'</span>],
              [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'PROPERTIES'</span>, value: <span class="code-quote">'/nfs_qa_pac/pac-pte/jenkins/env_properties/0010.TESTING-'</span> + STREAM_ID + <span class="code-quote">'-Headline.properties'</span>],
              [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'CHANGELIST'</span>, value: MX_CHANGELIST],
              [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'OPERATING_SYSTEM'</span>, value: OPERATING_SYSTEM],
              [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'DEBUG_NODE'</span>, value: HEAD_DEBUG_NODE],
              [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'DEBUG_APP_DIR'</span>, value: HEAD_DEBUG_APP_DIR]
             ],
             propagate: <span class="code-keyword">false</span>
            currentBuild.result = DeployClient.result
           } <span class="code-keyword">catch</span> (all) {
            echo <span class="code-quote">'Deploy Head Debug Client Failed'</span>
            <span class="code-keyword">return</span> 0
           }
           println(<span class="code-quote">"${STAGE_END} Deploying Client HEAD"</span>)
          }
         }
        }
       }
      }
     }    }
   }
</pre>
</div></div>
</pre>
</details>

</ul>
</details>

<p>I trigger jobs from inside the stage as shown below code sample.<br/>
 Noting that the <font color="#de350b"><b>Triggered Builds</b></font> section was working properly previously before you started allowing in Blue Ocean to display sequential stages inside a parallel one.</p>

<p><ins>It is very important and crucial to see the Triggered Builds within a stage while PROCESSING and not just when the stage is DONE.</ins></p>

<p> </p>

<pre>
stage(<span class="code-quote">'Pushing 2 Tests and 2 Debug jobs'</span>) {
    <span class="code-comment">//failFast <span class="code-keyword">true</span>
</span>    when {
     expression {
      Push2TEs == <span class="code-quote">'<span class="code-keyword">true</span>'</span>
     }
    }
    parallel {     stage(<span class="code-quote">'TE1'</span>) {
      stages {
       stage(<span class="code-quote">'PUSH TE1'</span>) {
options {
 retry(2)
 timeout(time: 3, unit: <span class="code-quote">'HOURS'</span>)
}
steps {
 script {
  timestamps {
   println(<span class="code-quote">"${STAGE_BEGIN} PUSH TE1"</span>)
   def TE1build = build job: <span class="code-quote">"TESTING/${TEST_LABEL1}/PAC_PTE/0010.TESTING-Headline"</span>,
    parameters: [
     [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'MX_VERSION'</span>, value: MX_VERSION],
     [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'MX_CHANGELIST'</span>, value: MX_CHANGELIST],
     [$class: <span class="code-quote">'BooleanParameterValue'</span>, name: <span class="code-quote">'Execute_benchmark'</span>, value: <span class="code-keyword">true</span>],
     [$class: <span class="code-quote">'BooleanParameterValue'</span>, name: <span class="code-quote">'Generate_traceability_reports'</span>, value: <span class="code-keyword">true</span>],
     [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'SETUPS'</span>, value: <span class="code-quote">'Custom'</span>],
     [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'Custom_Script_Before'</span>, value: <span class="code-quote">'/nfs_qa_pac/pac-pte/jenkins/getEnvProperties.sh'</span>],
     [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'Custom_Script_After'</span>, value: <span class="code-quote">'/nfs_qa_pac/pac-pte/jenkins/restoreEnv.sh'</span>]
    ],
    propagate: <span class="code-keyword">false</span>
   LogJobStatus(<span class="code-quote">"${TEST_LABEL1}"</span>, TE1build.result)
   currentBuild.result = TE1build.result
  }
 }
}
       }
       stage(<span class="code-quote">'Archive Logs'</span>) {
steps {
 script {
  timestamps {
   println(<span class="code-quote">"${STAGE_BEGIN} Archive ${TEST_LABEL1} Logs"</span>)
   <span class="code-comment">//sleep time: 1, unit: <span class="code-quote">'MINUTES'</span>
</span>   def archiveLogsTE1 = build job: <span class="code-quote">'Archiving'</span>,
    parameters: [
     [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'BENCH_MARK_NAME'</span>, value: TE1_BENCH_MARK_NAME],
     [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'VERSION'</span>, value: MX_VERSION],
     [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'CHANGELIST'</span>, value: MX_CHANGELIST],
     [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'APP_DIR'</span>, value: TE1_APP_DIR],
     [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'RESULTS_DIR'</span>, value: TE1_LOG_DIR],
     [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'KPI_DIR'</span>, value: TE1_WORKSPACE],
     [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'TPK'</span>, value: TPK],
     [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'JOB_ID'</span>, value: JOB_ID],
     [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'TEST_ID'</span>, value: TEST_LABEL1],
     [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'NODE'</span>, value: TE1_NODE]
    ],
    propagate: <span class="code-keyword">false</span>
   currentBuild.result = archiveLogsTE1.result
   LogJobStatus(<span class="code-quote">"Archive ${TEST_LABEL1} Logs"</span>, archiveLogsTE1.result)   println(<span class="code-quote">"${STAGE_END} Archive ${TEST_LABEL1} Logs"</span>)
  }
 }
}
       }
      }
     }     stage(<span class="code-quote">'TE2'</span>) {
      stages {
       stage(<span class="code-quote">'PUSH TE2'</span>) {
options {
 retry(2)
 timeout(time: 3, unit: <span class="code-quote">'HOURS'</span>)
}
steps {
 script {
  timestamps {
   println(<span class="code-quote">"${STAGE_BEGIN} PUSH TE2"</span>)
   def TE2build = build job: <span class="code-quote">"TESTING/${TEST_LABEL2}/PAC_PTE/0010.TESTING-Headline"</span>,
    parameters: [
     [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'MX_VERSION'</span>, value: MX_VERSION],
     [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'MX_CHANGELIST'</span>, value: MX_CHANGELIST],
     [$class: <span class="code-quote">'BooleanParameterValue'</span>, name: <span class="code-quote">'Execute_benchmark'</span>, value: <span class="code-keyword">true</span>],
     [$class: <span class="code-quote">'BooleanParameterValue'</span>, name: <span class="code-quote">'Generate_traceability_reports'</span>, value: <span class="code-keyword">true</span>],
     [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'SETUPS'</span>, value: <span class="code-quote">'Custom'</span>],
     [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'Custom_Script_Before'</span>, value: <span class="code-quote">'/nfs_qa_pac/pac-pte/jenkins/getEnvProperties.sh'</span>],
     [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'Custom_Script_After'</span>, value: <span class="code-quote">'/nfs_qa_pac/pac-pte/jenkins/restoreEnv.sh'</span>]
    ],
    propagate: <span class="code-keyword">false</span>
   LogJobStatus(<span class="code-quote">"PUSH TE2"</span>, TE2build.result)
   currentBuild.result = TE2build.result       
  }
 }
}
       }
       stage(<span class="code-quote">'Archive Logs'</span>) {
steps {
 script {
  timestamps {
   println(<span class="code-quote">"${STAGE_BEGIN} Archive ${TEST_LABEL2} Logs"</span>)
   <span class="code-comment">//sleep time: 1, unit: <span class="code-quote">'MINUTES'</span>
</span>   def archiveLogsTE2 = build job: <span class="code-quote">'Archiving'</span>,
    parameters: [
     [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'BENCH_MARK_NAME'</span>, value: TE2_BENCH_MARK_NAME],
     [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'VERSION'</span>, value: MX_VERSION],
     [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'CHANGELIST'</span>, value: MX_CHANGELIST],
     [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'APP_DIR'</span>, value: TE2_APP_DIR],
     [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'RESULTS_DIR'</span>, value: TE2_LOG_DIR],
     [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'TPK'</span>, value: TPK],
     [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'JOB_ID'</span>, value: JOB_ID],
     [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'TEST_ID'</span>, value: TEST_LABEL2],
     [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'NODE'</span>, value: TE2_NODE]
    ],
    propagate: <span class="code-keyword">false</span>
   currentBuild.result = archiveLogsTE2.result
   LogJobStatus(<span class="code-quote">"Archive ${TEST_LABEL2} Logs"</span>, archiveLogsTE2.result)   println(<span class="code-quote">"${STAGE_END} Archive ${TEST_LABEL2} Logs"</span>)
  }
 }
}
       }
      }
     }     stage(<span class="code-quote">'HEAD DEBUG'</span>) {
      stages {
       stage(<span class="code-quote">'PUSH HEAD DEBUG'</span>) {
options {
 retry(2)
 timeout(time: 2, unit: <span class="code-quote">'HOURS'</span>)
}
steps {
 script {
  timestamps {
   println(<span class="code-quote">"${STAGE_BEGIN} PUSH HEAD DEBUG"</span>)
   <span class="code-keyword">try</span> {
    <span class="code-comment">//STREAM_ID=<span class="code-quote">'DEBUG230820181643'</span>
</span>    def pushHeadDebugJob = build job: <span class="code-quote">"TESTING/${STREAM_ID}/PAC_PTE/0010.TESTING-Headline"</span>,
     parameters: [
      [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'MX_VERSION'</span>, value: MX_VERSION],
      [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'MX_CHANGELIST'</span>, value: MX_CHANGELIST],
      [$class: <span class="code-quote">'BooleanParameterValue'</span>, name: <span class="code-quote">'Execute_benchmark'</span>, value: <span class="code-keyword">false</span>],
      [$class: <span class="code-quote">'BooleanParameterValue'</span>, name: <span class="code-quote">'Generate_traceability_reports'</span>, value: <span class="code-keyword">false</span>],
      [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'SETUPS'</span>, value: <span class="code-quote">'Custom'</span>],
      [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'Custom_Script_Before'</span>, value: <span class="code-quote">'/nfs_qa_pac/pac-pte/jenkins/getEnvProperties.sh'</span>],
      [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'Custom_Script_After'</span>, value: <span class="code-quote">'/nfs_qa_pac/pac-pte/jenkins/restoreEnv.sh'</span>]
     ],
     propagate: <span class="code-keyword">false</span>
    currentBuild.result = pushHeadDebugJob.result   } <span class="code-keyword">catch</span> (all) {
    echo <span class="code-quote">'Pushing Head Debug job Failed'</span>
    <span class="code-keyword">return</span> 0
   }
   println(<span class="code-quote">"${STAGE_END} PUSH HEAD DEBUG"</span>)
  }
 }
}
       }
       stage(<span class="code-quote">'DEPLOY CLIENT HEAD'</span>) {
steps {
 script {
  timestamps {
   println(<span class="code-quote">"${STAGE_BEGIN} Deploying Client HEAD"</span>)
   <span class="code-keyword">try</span> {
    def DeployClient = build job: <span class="code-quote">'DeployClient'</span>,
     parameters: [
      [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'NODE'</span>, value: <span class="code-quote">'pacpte_pte02nt'</span>],
      [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'PROPERTIES'</span>, value: <span class="code-quote">'/nfs_qa_pac/pac-pte/jenkins/env_properties/0010.TESTING-'</span> + STREAM_ID + <span class="code-quote">'-Headline.properties'</span>],
      [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'CHANGELIST'</span>, value: MX_CHANGELIST],
      [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'OPERATING_SYSTEM'</span>, value: OPERATING_SYSTEM],
      [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'DEBUG_NODE'</span>, value: HEAD_DEBUG_NODE],
      [$class: <span class="code-quote">'StringParameterValue'</span>, name: <span class="code-quote">'DEBUG_APP_DIR'</span>, value: HEAD_DEBUG_APP_DIR]
     ],
     propagate: <span class="code-keyword">false</span>
    currentBuild.result = DeployClient.result
   } <span class="code-keyword">catch</span> (all) {
    echo <span class="code-quote">'Deploy Head Debug Client Failed'</span>
    <span class="code-keyword">return</span> 0
   }
   println(<span class="code-quote">"${STAGE_END} Deploying Client HEAD"</span>)
  }
 }
}
       }
      }
     }    }
   }
</pre>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>ekassis:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-53129?focusedId=349809&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-349809">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=olamy" class="user-hover" rel="olamy">olamy</a>, you can find below for sequential <b>Processing</b> stages, Blue Ocean is showing the <b>Triggered Builds</b> section.</p>

<p> </p>

<p><span class="image-wrap" style=""><img src="https://issues.jenkins.io/secure/attachment/44568/44568_image-2018-09-23-19-13-12-757.png" height="351" width="615" style="border: 0px solid black" /></span></p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=olamy" class="user-hover" rel="olamy">olamy</a>, you can find below for sequential <b>Processing</b> stages, Blue Ocean is showing the <b>Triggered Builds</b> section.</p>

<p> </p>

<p><span class="image-wrap" style=""><img src="https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/44568/image-2018-09-23-19-13-12-757.png" height="351" width="615" style="border: 0px solid black" /></span></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>ekassis:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-53129?focusedId=349811&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-349811">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=olamy" class="user-hover" rel="olamy">olamy</a>, a new bug detected, when the first stage in a sequential parallel pipeline is done, the Processing stages got <b>wrong status</b> and show only 1 stage processing even thought there are 4 stages processing.</p>

<p><span class="image-wrap" style=""><img src="https://issues.jenkins.io/secure/attachment/44569/44569_image-2018-09-23-20-42-44-112.png" height="265" width="649" style="border: 0px solid black" /></span></p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=olamy" class="user-hover" rel="olamy">olamy</a>, a new bug detected, when the first stage in a sequential parallel pipeline is done, the Processing stages got <b>wrong status</b> and show only 1 stage processing even thought there are 4 stages processing.</p>

<p><span class="image-wrap" style=""><img src="https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/44569/image-2018-09-23-20-42-44-112.png" height="265" width="649" style="border: 0px solid black" /></span></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/olamy.jpg" title="olamy's avatar" /> olamy:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-53129?focusedId=349818&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-349818">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=ekassis" class="user-hover" rel="ekassis">ekassis</a> please stop hijacking this issue. You reopen it and changing the description. This is totally wrong! it's a thread in a mailing list. So please open a new issue with correct description.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=ekassis" class="user-hover" rel="ekassis">ekassis</a> please stop hijacking this issue. You reopen it and changing the description. This is totally wrong! it's a thread in a mailing list. So please open a new issue with correct description.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/olamy.jpg" title="olamy's avatar" /> olamy:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-53129?focusedId=349819&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-349819">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>please open an other issue regarding the "Triggered Builds"</p>
</pre>
</details>

</ul>
</details>

<p>please open an other issue regarding the "Triggered Builds"</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>ekassis:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-53129?focusedId=349821&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-349821">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Sorry <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=olamy" class="user-hover" rel="olamy">olamy</a>, a new Jira issue opened:    <font color="#0747a6"> <b><a href="https://issues.jenkins.io/browse/JENKINS-53735" title="Parallel sequential stages not showing Triggered Builds while Processing" class="issue-link" data-issue-key="JENKINS-53735">JENKINS-53735</a></b></font></p>
</pre>
</details>

</ul>
</details>

<p>Sorry <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=olamy" class="user-hover" rel="olamy">olamy</a>, a new Jira issue opened:    <font color="#0747a6"> <b><a href="https://issues.jenkins.io/browse/JENKINS-53735" title="Parallel sequential stages not showing Triggered Builds while Processing" class="issue-link" data-issue-key="JENKINS-53735">JENKINS-53735</a></b></font></p>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `duplicates` from Jira: <a href="https://github.com/jenkinsci/blueocean-plugin/issues?q=is%3Aissue%20%22jira_issue_key%3DJENKINS-53311%22">JENKINS-53311</a>]</i>
